src/world_interface/base_motor.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+#include "data.h"
+#include "../control/JacobianVelController.h"
+#include "../utils/scheduler.h"
+
+using namespace std::chrono_literals;
+
+/**
+ * @namespace robot
+ * @brief Collection of functions for manipulating a motor.
+ */
+namespace robot {
+/**
+ * An abstract motor class
+ *
+ */
+class base_motor {
+public:
+	/**
+	 * @brief Constructor for base motor.
+	 *
+	 * @param motor The motor id to manipulate.
+	 * @param hasPosSensor Boolean to indicate if the motor has a position sensor.
+	 */
+	base_motor(robot::types::motorid_t motor, bool hasPosSensor)
```

Comment:
Can we move this implementation of this method to `base_motor.cpp`? We should minimize implementation in header files.